DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -267,18 +317,30 @@ protected override void ProcessFile(InstallFile file, XPathNavigator nav)
                     // This is the initial script when installing
                     this._installScript = file;
                 }
-                else if (file.Name.StartsWith("upgrade.", StringComparison.InvariantCultureIgnoreCase))
+                else if (file.Name.StartsWith("preupgrade.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("preupgrade", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    this._preUpgradeScript = file;
+                }
+                else if (file.Name.StartsWith("upgrade.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("postupgrade", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Should this type just be `"upgrade"`?  I would expect it to run last if it's `"postupgrade"`.

Reply:
Given that we have "preupgrade", "upgrade" becomes confusing. Currently "upgrade" is post-upgrade. I.e. after the versioned upgrade scripts have run.